lib/winrm-fs/core/file_transporter.rb

```diff
@@ -469,8 +474,12 @@ def stream_upload_file(src, dest, &block)
           chunks = 0
           bytes = 0
           elapsed = Benchmark.measure do
-            File.open(src, 'rb') do |io|
-              chunks, bytes = stream_upload(io, dest, &block)
+            if src.is_a?(Hash)
```

Comment:
if `src` remained a `StringIO` as suggested above, I think you could leave this addition out.

Reply:
Unfortunately `File.open` will error out if `src` is a `StringIO` object.